dlc-manager/src/lib.rs

```diff
@@ -77,27 +84,75 @@ impl Time for SystemTimeProvider {
 }
 
 /// Provides signing related functionalities.
-pub trait Signer {
-    /// Signs a transaction input
-    fn sign_psbt_input(
-        &self,
-        psbt: &mut PartiallySignedTransaction,
-        input_index: usize,
-    ) -> Result<(), Error>;
+pub trait ContractSigner: Clone {
+    /// Get the public key associated with the [`ContractSigner`].
+    fn get_public_key(&self) -> Result<PublicKey, Error>;
+    /// Returns the secret key associated with the [`ContractSigner`].
+    // todo: remove this method and add create_adaptor_signature to the trait
+    fn get_secret_key(&self) -> Result<SecretKey, Error>;
+}
+
+/// Simple sample implementation of [`ContractSigner`].
+#[derive(Debug, Copy, Clone)]
+pub struct SimpleSigner {
+    secret_key: SecretKey,
+}
+
+impl SimpleSigner {
+    /// Creates a new [`SimpleSigner`] from the provided secret key.
+    pub fn new(secret_key: SecretKey) -> Self {
+        Self { secret_key }
+    }
+}
+
+impl ContractSigner for SimpleSigner {
+    fn get_public_key(&self) -> Result<PublicKey, Error> {
+        Ok(self.secret_key.public_key(SECP256K1))
+    }
+
+    fn get_secret_key(&self) -> Result<SecretKey, Error> {
+        Ok(self.secret_key)
+    }
+}
+
+impl ContractSigner for SecretKey {
+    fn get_public_key(&self) -> Result<PublicKey, Error> {
+        Ok(self.public_key(SECP256K1))
+    }
+
+    fn get_secret_key(&self) -> Result<SecretKey, Error> {
+        Ok(*self)
+    }
+}
+
+/// Derives a [`ContractSigner`] from a [`ContractSignerProvider`] and a `contract_keys_id`.
+pub trait ContractSignerProvider {
+    /// A type which implements [`ContractSigner`]
+    type Signer: ContractSigner;
+
+    /// Create a keys id for deriving a `Signer`.
+    fn derive_signer_key_id(&self, is_offer_party: bool, temp_id: [u8; 32]) -> [u8; 32];
+
+    /// Derives the private key material backing a `Signer`.
+    fn derive_contract_signer(&self, key_id: [u8; 32]) -> Result<Self::Signer, Error>;
+
     /// Get the secret key associated with the provided public key.
+    ///
+    /// Only used for Channels.
```

Comment:
This will probably need to be split in the future, but fine if you don't do it now.